lib/build.js

```diff
@@ -152,24 +154,36 @@ class BuildModel extends BaseModel {
     start() {
         // Make sure that a pipeline and job is associated with the build
         return this.job
-            .then(job =>
-                job.pipeline.then(pipeline =>
+            .then((job) => {
+                let buildTimeout = DEFAULT_BUILD_TIMEOUT;
+
+                // buildTimeout is only set in executor-k8s or executor-k8s-vm
+                if (this[executor].kubernetes && this[executor].kubernetes.buildTimeout) {
+                    buildTimeout = this[executor].kubernetes.buildTimeout;
+                }
+
+                const annotations = job.permutations[0].annotations;
+                const jwtExpiresSec = annotations[ANNOTATION_BUILD_TIMEOUT]
+                    ? annotations[ANNOTATION_BUILD_TIMEOUT] * 60
+                    : buildTimeout * 60;
+
+                return job.pipeline.then(pipeline =>
                     this[executor].start({
-                        annotations: job.permutations[0].annotations,
+                        annotations,
                         apiUri: this[apiUri],
                         buildId: this.id,
                         container: this.container,
                         token: this[tokenGen](this.id, {
                             isPR: job.isPR(),
                             jobId: job.id,
                             pipelineId: pipeline.id
-                        }, pipeline.scmContext)
+                        }, pipeline.scmContext, jwtExpiresSec)
```

Comment:
need a unit test case update for checking this new value being passed.

Reply:
I added the unit test.